src/main/java/com/kodilla/ecommercee/service/ProductService.java

```diff
@@ -0,0 +1,58 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.domain.dto.ProductDto;
+import com.kodilla.ecommercee.exception.GroupNotFoundException;
+import com.kodilla.ecommercee.exception.ProductNotFoundException;
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class ProductService {
+    private final ProductRepository productRepository;
+    private final GroupRepository groupRepository;
+
+    @Autowired
+    public ProductService(ProductRepository productRepository, GroupRepository groupRepository) {
+        this.productRepository = productRepository;
+        this.groupRepository = groupRepository;
+    }
+
+    public List<Product> getProductList() {
+        return productRepository.findAll();
+    }
+
+    public Product getProductById(long id) {
+        return productRepository.findById(id).orElseThrow(ProductNotFoundException::new);
+    }
+
+    public void addProduct(Product product) throws GroupNotFoundException {
+        productRepository.save(product);
+    }
```

Comment:
Można to tylko kwestia zapisu innego ale czy ta metoda nie powinna być typu Product i zwracać to samo.

Reply:
Mogłaby, ale ta metoda jest używana tylko wewnątrz metody addProduct ProductControllera, która jest typu void, więc chyba nie ma potrzeby.

Reply:
Ok, tak jak mówię kwestia zapisu.